.github/workflows/ci-cd.yaml

```diff
@@ -94,8 +94,88 @@ jobs:
       - name: trento checks ID sanity test
         run: python3 hack/id_checker.py
 
+  build-container-images:
+    runs-on: ubuntu-20.04
+    needs: [test-binary, test-checks]
+    strategy:
+      matrix:
+        image:
+          - trento-web
+          - trento-runner
+    permissions:
+      contents: read
+      packages: write
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_REPOSITORY: ghcr.io/${{ github.repository_owner }}/${{ matrix.image }}
+      IMAGE_TAG: "${{ (github.event_name == 'release' && github.event.release.tag_name) || (github.event_name == 'push' && github.ref_name == 'main' && 'rolling') || github.sha }}"
+      PUSH: "${{ github.event_name == 'release' || (github.event_name == 'push' && github.ref_name == 'main') || github.event_name == 'workflow_dispatch' }}"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: docker/setup-buildx-action@v1
+      - name: Log in to the Container registry
+        if: ${{ env.PUSH }}
```

Comment:
I did notice this also on my orignial PR, but for some reason this `if` is not doing anything.
I would suggest the following:
- try to remove the `${{}}` notation and set it to just `if: env.PUSH`, and if see it still gets run (it shouldn't. unless it is a push on main, a release, or a manual trigger)
- if the above doesn't change anything, then remove this `if` condition and let this step always run. In this case, you might want to also remove the `PUSH` environment variable and move the conditional expression in the below `push` field of the `build-and-push` action.

Reply:
Do we need this at the `step` level or could we move it to the job level? It seems to me that what we want is the whole job to get skipped (not that specific step only)

Reply:
no, this can't be at the job level, because currently the same job has two unrelated group of steps:
- log in the registry, build the images, and push them to the registry. These only happens if it's a release or a merge to main.
- build the images (again, but there is cache, so takes almost no time), and export them as an artifact. This happens regardless of whether or not it's a release or a merge to main.

If the `if` condition at the step level doesn't work, leave it out and let the log-in happen all times, even if we only need for the push to registry.

Otherwise, you could try to split this entire thing into two jobs, maybe it'll be cleaner.

Reply:
I've splitted the thing into two jobs and the `if` now works correctly. It builds the container assets but doesn't push the image unless it needs to. Could you check it again?

Reply:
ok, now we have two new minor problems:
- naming: the difference between the two jobs is not super clear. I suggest something like `build-and-push-containers` and `build-and-export-containers`, but feel free to propose something else.
- dependencies: `smoke-test-container-images` now depends on `build-and-export-containers`, while `deploy-server` depends on both `smoke-test-container-images` and `build-and-push-containers`. Make sure the flow is correct.

Reply:
should be ok now!